pkg/storage/mysql/mysql_test.go

```diff
@@ -1,21 +1,113 @@
-package mysql_test
+package mysql
 
 import (
+	"context"
 	"testing"
+	"time"
 
-	"github.com/openfga/openfga/pkg/storage/common"
-	"github.com/openfga/openfga/pkg/storage/mysql"
+	sq "github.com/Masterminds/squirrel"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/sqlcommon"
 	"github.com/openfga/openfga/pkg/storage/test"
 	storagefixtures "github.com/openfga/openfga/pkg/testfixtures/storage"
+	"github.com/openfga/openfga/pkg/tuple"
 	"github.com/stretchr/testify/require"
+	openfgav1 "go.buf.build/openfga/go/openfga/api/openfga/v1"
 )
 
 func TestMySQLDatastore(t *testing.T) {
 	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
 
 	uri := testDatastore.GetConnectionURI()
-	ds, err := mysql.New(uri, common.NewConfig())
+	ds, err := New(uri, sqlcommon.NewConfig())
 	require.NoError(t, err)
 	defer ds.Close()
 	test.RunAllTests(t, ds)
 }
+
+// TestReadEnsureNoOrder asserts that the read response is not ordered by ulid
+func TestReadEnsureNoOrder(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
+
+	uri := testDatastore.GetConnectionURI()
+	ds, err := New(uri, sqlcommon.NewConfig())
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	defer ds.Close()
```

Comment:
Please move the defer directly below the construction.

```go
ds, err : New(...)
defer ds.Close()

require.NoError(t, err)
```

Reply:
done

Reply:
Actually, lint suggests we need to defer close after checking for error

Reply:
That's fine if the linter is warning.

```go
ds, err := New(...)
require.NoError(t, err)
defer ds.Close()

ctx := context.Background()
```

keeping the defer as close to the instantiation as possible